Fix/site header layout fixes

yaelirub opened the pull request:
Fixes #14617 and #14618 
To test:

Site name trailing 
1. Go to reader
2. Tap on a site name in the post card (a site with a long name like theinquisitivebee.co.uk)
3. See that the site name doesn't get cut and has a 16 trailing to superview.

Empty site description 
1. Go to reader
2. Tap on a site name in the post card (a site that doesn't have description like aerych.blog)
3. See that the header bottom is 16 to image bottom .

Make sure the header looks as expected with 1 liner and long description

![Simulator Screen Shot - iPhone 11 Pro - 2020-08-11 at 17 52 07](https://user-images.githubusercontent.com/1335657/89964965-64f35200-dc00-11ea-8050-1a406b3aa5da.png)
![Simulator Screen Shot - iPhone 11 Pro - 2020-08-11 at 18 19 02](https://user-images.githubusercontent.com/1335657/89964973-691f6f80-dc00-11ea-82c9-9d65c1fd75c2.png)



PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

yaelirub committed:
Fix layout issues in site header
- Updated trailing constraints for site name, url and detail labels
- Updated bottom constraint for description label

yaelirub committed:
updating description label top constraint if it's empty

emilylaguna reviewed (changes requested):
Heya! 👋 The wrapping of the site header is working well, but I noticed a few issues:

When I tap the follow button on a site that does not have a description, the application crashes:
```
Fatal error: Unexpectedly found nil while implicitly unwrapping an Optional value: file /Users/emilylaguna/Desktop/Work/WordPress-iOS/WordPress/Classes/ViewRelated/Reader/ReaderSiteStreamHeader.swift, line 88
2020-08-12 12:27:41.702938-0400 WordPress[23883:194151] Fatal error: Unexpectedly found nil while implicitly unwrapping an Optional value: file /Users/emilylaguna/Desktop/Work/WordPress-iOS/WordPress/Classes/ViewRelated/Reader/ReaderSiteStreamHeader.swift, line 88
```
<img src="https://user-images.githubusercontent.com/793774/90041697-82670100-dc7e-11ea-8415-8ffd74450dd3.gif" width="30%"/>

---

I am not seeing the 16px right margin as indicated here: https://github.com/wordpress-mobile/WordPress-iOS/issues/14618 when the site title is just long enough

<img width="40%" alt="Screen Shot 2020-08-12 at 12 10 26 PM" src="https://user-images.githubusercontent.com/793774/90039581-b260d500-dc7b-11ea-80d0-1fd54205e25f.png">

---

The gap between the site icon / followers label seems pretty large, is this intended?

<img width="40%" alt="Screen Shot 2020-08-12 at 12 17 21 PM" src="https://user-images.githubusercontent.com/793774/90040379-b6412700-dc7c-11ea-9cd4-310cb91b1963.png">

---

When in a RTL language the follow title overlaps the follow icon:

<img width="20%" alt="Screen Shot 2020-08-12 at 12 24 21 PM" src="https://user-images.githubusercontent.com/793774/90041097-ab3ac680-dc7d-11ea-903d-7e363ab0d2e6.png">

---

If I tap and hold on the follow button the title text disappears

<img src="https://user-images.githubusercontent.com/793774/90042126-1933bd80-dc7f-11ea-9e81-113bca336f7b.gif" width="30%" />

emilylaguna reviewed (changes requested):
I did some further testing with a small screen device and encountered a overlap issue:

<img width="432" alt="Screen Shot 2020-08-12 at 1 06 12 PM" src="https://user-images.githubusercontent.com/793774/90045219-919c7d80-dc83-11ea-8147-275ce7fd09b0.png">

yaelirub commented:
>I did some further testing with a small screen device and encountered a overlap issue:

Not sure how you get to this screen but I see:
![Simulator Screen Shot - iPhone SE (2nd generation) - 2020-08-12 at 10 21 09](https://user-images.githubusercontent.com/1335657/90047250-7aab5a80-dc86-11ea-8218-b564fb710fa6.png)

It's weird because you don't even have the back title in the navigation?

yaelirub committed:
Updated constraints when description label has no text
Fixing issues found in code review

yaelirub commented:
Hey @emilylaguna , thanks for the through review 🙏 
>When I tap the follow button on a site that does not have a description, the application crashes:

I couldn't reproduce. Maybe because I removed the deactivation of the constraint. 

>I am not seeing the 16px right margin as indicated here: #14618 when the site title is just long enough

Couldn't reproduce. 

>The gap between the site icon / followers label seems pretty large, is this intended?

Its20px by design:
<img width="952" alt="Screen Shot 2020-08-12 at 12 19 33 PM" src="https://user-images.githubusercontent.com/1335657/90058456-99b1e880-dc96-11ea-927f-d1fb7246029b.png">

>I did some further testing with a small screen device and encountered a overlap issue:

Fixed in https://github.com/wordpress-mobile/WordPress-iOS/pull/14622/commits/e7d5c9c3f85bb7520a360b79ac7731384f27a2eb

Ready for another look

yaelirub committed:
Updated spacing from 20 to 16 following design mock update

yaelirub committed:
Updated spacing between follow count and button to 8

emilylaguna commented on WordPress/Classes/ViewRelated/Reader/ReaderSiteStreamHeader.swift:
```diff
@@ -85,8 +85,7 @@ fileprivate func > <T: Comparable>(lhs: T?, rhs: T?) -> Bool {
         if siteTopic.siteDescription.isEmpty {
-            descriptionLableTopConstraint.isActive = false
-            descriptionLabel.topAnchor.constraint(equalTo: avatarImageView.bottomAnchor, constant: 0.0).isActive = true
+            descriptionLableTopConstraint.constant = 0.0
```
```suggestion
            descriptionLabelTopConstraint.constant = 0.0
```

yaelirub replied:
updated here 75ca4e7

emilylaguna reviewed (changes requested):
This is working for me now, 1 final typo fix and we'll be ready to :shipit:

yaelirub committed:
Fixed variable typo

emilylaguna reviewed (approved):
:shipit:
